Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -140,12 +140,19 @@ void WarpXOpenPMDPlot::GetFileName(std::string& filename)
 
 void WarpXOpenPMDPlot::SetStep(int ts)
 {
-  if (ts < 0)
-    return;
+  AMREX_ALWAYS_ASSERT_WITH_MESSAGE(ts >= 0 , "openPMD iterations are unsigned");
+
+  if (m_CurrentStep >= ts) {
+      // note m_Series is reset in Init(), so using m_Series->iterations.contains(ts) is only able to check the
+      // last written step in m_Series's life time, but not other earlier written steps by other m_Series
+      std::string warnMsg = " Warning from openPMD writer: Already written iteration:"+std::to_string(ts);
+      std::cout<<warnMsg<<std::endl;
```

Comment:
please include `<iostream>` in this file :)

Reply:
It is included in AMReX_Utility.H which is included in WarpXOpenPMD.H

Reply:
Yes, that's the reason why this TU currently compiles.

But since we cannot assume the includes in transitive dependencies stay that way, this is fragile. It is therefore [good practice in C/C++ to include used headers in .cpp files](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md).

Reply:
@ax3l
Axel,  the LGTM fails for C# build error, when you have a chance, can you see why?

Reply:
Taking a look if this an issue already in `dev` or just needs to be re-run :)